bp_common/src/include/bp_common_me_if.vh

```diff
@@ -183,105 +181,123 @@ typedef enum bit [1:0]
-  ,e_lce_cmd_set_tag         = 3'b100
-  ,e_lce_cmd_set_tag_wakeup  = 3'b101
-  ,e_lce_cmd_invalidate_tag  = 3'b110
-  ,e_lce_cmd_uc_st_done      = 3'b111
+  e_lce_cmd_sync             = 4'b0000
+  ,e_lce_cmd_set_clear       = 4'b0001
+  ,e_lce_cmd_transfer        = 4'b0010
+  ,e_lce_cmd_writeback       = 4'b0011
+  ,e_lce_cmd_set_tag         = 4'b0100
+  ,e_lce_cmd_set_tag_wakeup  = 4'b0101
+  ,e_lce_cmd_invalidate_tag  = 4'b0110
+  ,e_lce_cmd_uc_st_done      = 4'b0111
+  // 4'b1000 - 4'b1111 reserved / custom
 } bp_cce_lce_cmd_type_e;
 
 `define bp_cce_lce_cmd_type_width $bits(bp_cce_lce_cmd_type_e)
 
 /*
- * bp_cce_coh_mesi_e defines the coherence states for a MESI protocol
- * e_MESI_I means the block is invalid and the LCE does not have read or write permissions
- * e_MESI_S means the block is valid and the LCE has read permissions
- * e_MESI_E means the block is valid and the LCE has read and write permissions. The block may or
- *   may not be dirty in the LCE.
- * e_MESI_M has the same meaning as e_MESI_E, but the CCE knows the block is dirty
- */
-typedef enum bit [1:0] 
-{
-  e_MESI_I                   = 2'b00
-  ,e_MESI_S                  = 2'b01
-  ,e_MESI_E                  = 2'b10
-  ,e_MESI_M                  = 2'b11
-} bp_cce_coh_mesi_e;
-
-/*
- * bp_cce_coh_vi_e defines the coherence states for a Valid/Invalid style protocol.
- * In VI, the V state is equivalent to e_MESI_E, and I is the same as e_MESI_I.
+ * bp_cce_coh_states_e defines the coherence states available in BlackParrot. Each bit represents
+ * a property of the cache block as defined below:
+ * 0: Shared (not Exclusive)
+ * 1: Owned
+ * 2: Potentially Dirty
+ *
+ * These properties are derived from "A Primer on Memory Consistency and Cache Coherence", and
+ * they allow an easy definition for the common MOESIF coherence states.
  */
-typedef enum bit [1:0] 
-{
-  e_VI_I                     = 2'b00
-  ,e_VI_V                    = 2'b10
-} bp_cce_coh_vi_e;
-
-typedef union packed 
+typedef enum bit [2:0] 
 {
-  bp_cce_coh_vi_e            vi;
-  bp_cce_coh_mesi_e          mesi;
-}  bp_cce_coh_u;
+  e_COH_I                   = 3'b000 // Invalid
+  ,e_COH_S                  = 3'b001 // Shared - clean, not owned, shared (not exclusive)
+  ,e_COH_E                  = 3'b010 // Exclusive - clean, owned, not shared (exclusive)
+  ,e_COH_F                  = 3'b011 // Forward - clean, owned, shared (not exclusive)
+  //                          3'b100 - potentially dirty, not owned, not shared (exclusive)
+  //                          3'b101 - potentially dirty, not owned, shared (not exclusive)
+  ,e_COH_M                  = 3'b110 // Modified - potentially dirty, owned, not shared (exclusive)
+  ,e_COH_O                  = 3'b111 // Owned - potentially dirty, owned, shared (not exclusive)
+} bp_coh_states_e;
+
+`define bp_coh_shared_bit 0
+`define bp_coh_owned_bit 1
+`define bp_coh_dirty_bit 2
+
+`define bp_coh_bits $bits(bp_coh_states_e)
```

Comment:
Nit, but I don't think we need a macro for width on enums, since they can never be parameterizable

Reply:
(so $bits will always be sufficient)